File: yizhigou-service-interface/src/main/java/com/yizhigou/sellergoods/service/BrandService.java
package com.yizhigou.sellergoods.service;

import com.yizhigou.entity.PageResult;
import com.yizhigou.entity.Result;
import com.yizhigou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    List<TbBrand> getAllBrands();

    PageResult findPage(TbBrand tbBrand, int pageNum, int pageSize);

    Result add(TbBrand tbBrand);

    Result update(TbBrand tbBrand);

    TbBrand findOne(long id);

    void delete(Long[] ids);

    List<Map> selectBrandOptionList();
}

File: yizhigou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService", function ($http) {

    this.search = function (page,rows,searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&rows=' + rows,searchEntity);
    }

    this.findOne = function (id) {
        return $http.post('../brand/findOne.do?id=' + id);
    }

    this.delete = function (ids) {
        return $http.post('../brand/delete.do?ids=' + ids);
    }

    this.add = function (entity) {
        return $http.post("../brand/add.do" , entity);
    }

    this.update = function (entity) {
        return $http.post("../brand/update.do" , entity);
    }

    //请求品牌列表
    this.selectBrandOptionList = function () {
        return $http.post("../typeTemplate/selectBrandOptionList.do");
    }
})